client/txn.go

```diff
@@ -40,34 +40,34 @@ var (
 )
 
 type txnSender struct {
-	*Txn
+	*txn
 }
 
 func (ts *txnSender) Send(ctx context.Context, call Call) {
 	// Send call through wrapped sender.
-	call.Args.Header().Txn = &ts.txn
+	call.Args.Header().Txn = &ts.txn.txn
 	ts.wrapped.Send(ctx, call)
-	ts.txn.Update(call.Reply.Header().Txn)
+	ts.txn.txn.Update(call.Reply.Header().Txn)
 
 	if err, ok := call.Reply.Header().GoError().(*proto.TransactionAbortedError); ok {
 		// On Abort, reset the transaction so we start anew on restart.
-		ts.txn = proto.Transaction{
-			Name:      ts.txn.Name,
-			Isolation: ts.txn.Isolation,
+		ts.txn.txn = proto.Transaction{
+			Name:      ts.txn.txn.Name,
+			Isolation: ts.txn.txn.Isolation,
 			Priority:  err.Txn.Priority, // acts as a minimum priority on restart
 		}
 	}
 }
 
-// Txn provides serial access to a KV store via Run and parallel
+// txn provides serial access to a KV store via Run and parallel
 // access via Prepare and Flush. On receipt of
 // TransactionRestartError, the transaction epoch is incremented and
 // error passed to caller. On receipt of TransactionAbortedError, the
 // transaction is re-created and the error passed to caller.
 //
-// A Txn instance is not thread safe.
-type Txn struct {
-	kv           KV
+// A txn instance is not thread safe.
+type txn struct {
```

Comment:
So type `txn` contains a `proto.Transaction` also named `txn`, as well as a `txnSender` which embeds a `*txn` (and we also have a type `Tx`). This is all pretty confusing (and it appears to be confusing the compiler as well, if I'm interpreting the `txn.` additions above correctly). Let's rename `txn.txn` to `txn.txnProto`. Can we move the `Send` method to type `txn` and eliminate the `txnSender` type?

Is there any hope of merging this with the exported `Tx` type? If not is there a better name to express the distinction than `Tx` versus `txn`? Maybe `kvTxn`? For that matter, I'd also rename `Tx` to `Txn` now that the name is available - we use "txn" frequently but don't use "tx" anywhere else.